sympy/functions/elementary/trigonometric.py

```diff
@@ -1219,9 +1230,18 @@ def _eval_is_real(self):
     def _eval_is_finite(self):
         arg = self.args[0]
 
+        if arg.is_real and (arg / pi - S.Half).is_integer is False:
+            return True
+
         if arg.is_imaginary:
             return True
 
+    def _eval_is_complex(self):
+        arg = self.args[0]
+
+        if arg.is_real and (arg / pi - S.Half).is_integer is False:
+            return True
```

Comment:
If we had extended_complex then methods like this wouldn't be needed. We could just say that extended_complex is True and let `_eval_is_finite` imply complex.